test/controllers/nonsipp/memberpayments/UnallocatedContributionCYAControllerSpec.scala

```diff
@@ -88,4 +116,64 @@ class UnallocatedContributionCYAControllerSpec extends ControllerBaseSpec {
       )
     }
   }
+
+  "UnallocatedContributionCYAController in view only mode" - {
+
+    val currentUserAnswers = defaultUserAnswers
+      .unsafeSet(UnallocatedEmployerAmountPage(srn), money)
+      .unsafeSet(CompilationOrSubmissionDatePage(srn), submissionDateTwo)
+      .unsafeSet(FbVersionPage(srn), "002")
+
+    val previousUserAnswers = currentUserAnswers
+      .unsafeSet(FbVersionPage(srn), "001")
+      .unsafeSet(CompilationOrSubmissionDatePage(srn), submissionDateOne)
+      .unsafeSet(UnallocatedEmployerAmountPage(srn), money)
+
+    act.like(
+      renderView(onPageLoadViewOnly, userAnswers = currentUserAnswers, optPreviousAnswers = Some(previousUserAnswers)) {
+        implicit app => implicit request =>
+          injected[CYAWithRemove].apply(
+            viewModel(
+              srn,
+              schemeName,
+              money,
+              ViewOnlyMode,
+              viewOnlyUpdated = false,
+              optYear = Some(yearString),
+              optCurrentVersion = Some(submissionNumberTwo),
+              optPreviousVersion = Some(submissionNumberOne),
+              compilationOrSubmissionDate = Some(submissionDateTwo)
+            )
+          )
+      }.before(mockTaxYear(dateRange))
+        .withName("OnPageLoadViewOnly renders ok with no changed flag")
```

Comment:
I think it would be good to also have a test for the page rendering correctly with the Updated flag, what do you think?